### PR TITLE
Add batching per masJobRequest

### PR DESCRIPTION
**Pull request by samleeflang**
Created a MasJobRequest object which has the id and the batching attributes of the request for a MAS.
This also creates the possibility to add more attributes in the future.

### CODE REVIEW

**Review by southeo — COMMENTED** [review]
:microbe:

File: src/main/java/eu/dissco/backend/controller/BaseController.java [hunk 1]
```diff
@@ -30,16 +35,16 @@ protected String getPath(HttpServletRequest request) {
...
     if (!requestBody.data().type().equals("MasRequest")) {
       throw new ConflictException();
     }
     if (requestBody.data().attributes().get("mass") == null) {
       throw new IllegalArgumentException();
     }
-    return Arrays.asList(
-        mapper.treeToValue(requestBody.data().attributes().get("mass"), String[].class));
+    var list = mapper.convertValue(requestBody.data().attributes().get("mass"), LIST_TYPE);
+    return list.stream().collect(Collectors.toMap(MasJobRequest::masId, Function.identity()));
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/backend/controller/BaseController.java:47]: Good point, we discussed this and the object is more extendable this way.

File: src/main/java/eu/dissco/backend/service/MachineAnnotationServiceService.java [hunk 2]
```diff
@@ -112,12 +116,13 @@ public JsonApiListResponseWrapper scheduleMass(JsonNode flattenObjectData, List<
...
   }
 
-  private void validateBatchingRequest(Boolean batchingRequested,
+  private void validateBatchingRequest(Map<String, MasJobRequest> mass,
       List<MachineAnnotationServiceRecord> masRecords) throws ConflictException {
-    if (Boolean.FALSE.equals(batchingRequested)) {
-      return;
-    }
     for (var masRecord : masRecords) {
+      var batchingRequested = mass.get(masRecord.id()).batching();
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/backend/service/MachineAnnotationServiceService.java:122]: Good point, we discussed this and the object is more extendable.

File: src/main/java/eu/dissco/backend/service/MachineAnnotationServiceService.java [hunk 3]
```diff
@@ -79,23 +81,25 @@ public JsonApiListResponseWrapper getMassForObject(JsonNode jsonNode, String pat
...
-  public JsonApiListResponseWrapper scheduleMass(JsonNode flattenObjectData, List<String> mass,
-      String path, Object object, String targetId, String orcid, MjrTargetType targetType,
-      boolean batchingRequested)
-      throws ConflictException {
-    var masRecords = repository.getMasRecords(mass);
-    validateBatchingRequest(batchingRequested, masRecords);
+  public JsonApiListResponseWrapper scheduleMass(JsonNode flattenObjectData,
+      Map<String, MasJobRequest> masRequests, String path, Object object, String targetId,
+      String orcid,
+      MjrTargetType targetType) throws ConflictException {
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/backend/service/MachineAnnotationServiceService.java]: Should be fixed

**Review by southeo — APPROVED** [review]
:microscope:

**Pull request merged**